src/components/web3/connectors.ts

```diff
@@ -71,6 +71,19 @@ export const injected = new InjectedConnector({
   supportedChainIds: supportedChains.map((chain) => chain.id),
 })
 
+// get our standard supported chain and try to append the possibly unknown chain the user is on
+export const getInjectedConnector = async () => {
+  const { ethereum } = window as any
+  const currentProvider = new providers.Web3Provider(ethereum)
+  const chainId = (await currentProvider.getNetwork()).chainId
+  // append the current chain to the supported chains.
+  // can push duplicate ids, when user is on supported chain but that does not seem to make any problems.
+  const chains = [...supportedChains.map((chain) => chain.id), chainId]
```

Comment:
This would make it unique
```suggestion
  const chains = Array.from(new Set([...supportedChains.map((chain) => chain.id), chainId]))
```